### ISSUE TITLE
Runtime linking with OpenCL ICD

### ISSUE THREAD

**Issue by xmrig** [xmrig/xmrig-amd#145]
In v2.7.3 added runtime linking with OpenCL ICD (OpenCL.dll/libOpenCL.so)

Benefits:
* Miner does not directly depend on the library, OpenCL ICD loaded in runtime.
* **AMD APP SDK not required anymore** to build miner from source.
* Possible to specify custom path to OpenCL ICD via option `opencl-loader`.
* Automatic version detection (1.2 or 2.0), no more errors related to `clCreateCommandQueueWithProperties` function. 

Source now available in [dev](https://github.com/xmrig/xmrig-amd/tree/dev) branch `git checkout dev`.

**Comment by xmrig** [comment]
Windows binary for checking possible compatibility issues
```
ae920d9f714c01fd969f90fe31cff216ee4cb016f7eb4843b853652b3e7f18dd xmrig-amd-2.7.3-dev-win64.zip/xmrig-amd.exe
```
[xmrig-amd-2.7.3-dev-win64.zip](https://github.com/xmrig/xmrig-amd/files/2203096/xmrig-amd-2.7.3-dev-win64.zip)

**Comment by xmrig** [comment]
Implemented in v2.7.3-beta.
